include/seastar/core/gate.hh

```diff
@@ -233,42 +258,46 @@ public:
         holder& operator=(holder&& x) noexcept {
             if (&x != this) {
                 release();
-                _g = std::exchange(x._g, nullptr);
+                _g = std::move(x).release_gate();
```

Comment:
Why std::move() here? It's not harmful, just odd.

Reply:
It's there to show that x is going away.
We can have `release_gate` operate on a lvalue but the fact it doesn't leave the gate, like `release` make it more suitable to operate on a rvalue since the released gate needs to be picked up and be tracked.

Reply:
That's the way I started, but then I can't call it in `release()`.
Since it's private we can get away with just documenting any gotchas
and keep the call sites minimal. WDYT?